sys/hacl/build.rs

```diff
@@ -25,34 +30,42 @@ fn includes(home_dir: &Path) -> Vec<String> {
 }
 
 fn append_simd128_flags(flags: &mut Vec<String>) {
-    // Platform detection
-    if cfg!(simd128) {
-        flags.push("-DSIMD128".to_string());
-        if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
-            flags.push("-msse4.1".to_string());
-            flags.push("-msse4.2".to_string());
-            flags.push("-mavx".to_string());
-        }
+    if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
+        flags.push("-msse4.1".to_string());
+        flags.push("-msse4.2".to_string());
+        flags.push("-mavx".to_string());
     }
 }
 
 fn append_simd256_flags(flags: &mut Vec<String>) {
-    // Platform detection
-    if cfg!(simd256) {
-        flags.push("-DSIMD256".to_string());
-        flags.push("-mavx2".to_string());
-    }
+    flags.push("-mavx2".to_string());
 }
 
 #[cfg(not(windows))]
-fn create_bindings(home_dir: &Path) {
-    let mut clang_args = includes(home_dir);
-    append_simd128_flags(&mut clang_args);
-    append_simd256_flags(&mut clang_args);
+fn create_bindings(platform: Platform, home_dir: &Path) {
+    let mut clang_args = includes(home_dir, "-I");
 
-    let bindings = bindgen::Builder::default()
+    let mut bindings = bindgen::Builder::default();
+    bindings = bindings
         // Header to wrap HACL headers
-        .header("c/include/hacl.h")
+        .header("c/include/hacl.h");
+
+    if platform.simd128 {
+        append_simd128_flags(&mut clang_args);
+        clang_args.push("-DSIMD128".to_string());
```

Comment:
These are still used in the hand-written `config.h`. But we should probably set the right defines here instead.

Reply:
Here is what happens when we don't add `-DSIMD128/-DSIMD256` flags
libhacl128 built first and so it includes `libintvector.h` with only `HACL_CAN_COMPILE_VEC128` defined, see `hacl128.h` and `config.h`. when libhacl256 is building, it has `libintvector.h` with no `HACL_CAN_COMPILE_VEC256` defined and the build fails.

Reply:
Ok, I still think we can simplify this. But let's not do it right now if it's working as is.